src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const objectLog = [];
+  const objectToModify = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can destructure all needed props from action (type, extraData, keysToRemove)